src/cache.ts

```diff
@@ -0,0 +1,139 @@
+import { CacheDelResponse, CacheGetResponse, CacheSetOptions, CacheSetResponse } from "./types";
+import { CacheClient } from "./proto/server/v1/cache_grpc_pb";
+import {
+	DelRequest as ProtoDelRequest,
+	GetRequest as ProtoGetRequest,
+	KeysRequest as ProtoKeysRequest,
+	SetRequest as ProtoSetRequest,
+} from "./proto/server/v1/cache_pb";
+import { Utility } from "./utility";
+import { TigrisClientConfig } from "./tigris";
+
+export class Cache {
+	private readonly _projectName: string;
+	private readonly _cacheName: string;
+	private readonly _cacheClient: CacheClient;
+	private readonly _config: TigrisClientConfig;
+
+	constructor(
+		projectName: string,
+		cacheName: string,
+		cacheClient: CacheClient,
+		config: TigrisClientConfig
+	) {
+		this._projectName = projectName;
+		this._cacheName = cacheName;
+		this._cacheClient = cacheClient;
+		this._config = config;
+	}
+
+	public getCacheName(): string {
+		return this._cacheName;
+	}
+
+	/**
+	 * Sets the key with value. It will override the value if already exists
+	 * @param key
+	 * @param value
+	 * @param options - optionally set params.
```

Comment:
Could you add example invocations of the function, with and without options? Could you also do it for the get, del and keys functions?

Reply:
sure added in description. It is also present in test code.

Reply:
Great, thanks for adding to the description.